docs/directives.md

```diff
@@ -2,35 +2,49 @@
 title: Directives
 ---
 
-TypeGraphQL provides basic support for [GraphQL directives](https://www.apollographql.com/docs/graphql-tools/schema-directives/).
+[GraphQL directives](https://www.apollographql.com/docs/graphql-tools/schema-directives/) though the syntax might remind the TS decorators:
 
 > A directive is an identifier preceded by a @ character, optionally followed by a list of named arguments, which can appear after almost any form of syntax in the GraphQL query or schema languages.
 
-Here's a modified example from the GraphQL draft specification that illustrates several of these possibilities:
+But in fact, they are a purely SDL (Schema Definition Language) feature that allows you to put some metadata for selected type or its field:
 
 ```graphql
-directive @deprecated(
-  reason: String = "No longer supported"
-) on OBJECT | FIELD_DEFINITION
-
-@deprecated(reason: "Use `NewType`.")
-type Foo {
-  field: String
+type Foo @auth(requires: USER) {
+  field: String!
 }
 
 type Bar {
-  newField: String
-  oldField: String @deprecated(reason: "Use `newField`.")
+  field: String! @auth(requires: USER)
 }
 ```
 
-With type-graphql, you can add directives on fields and objects with `@Directive(nameOrDeclaration: string)` as in this example:
+That metadata can be read in runtime to modify the structure and behavior of a GraphQL schema to support reusable code and tasks like authentication, permission, formatting, and plenty more. They are also really useful for some external services like [Apollo Cache Control](https://www.apollographql.com/docs/apollo-server/performance/caching/#adding-cache-hints-statically-in-your-schema) or [Apollo Federation](https://www.apollographql.com/docs/apollo-server/federation/introduction/#federated-schema-example).
+
+**TypeGraphQL** of course provides a support for using the schema directives via the `@Directive` decorator.
+
+## Usage
+
+### Declaring in schema
+
+Basically, there are two supported ways of declaring the usage of directives:
+
+- string based - just like in SDL, with the `@` syntax:
+
+```typescript
+@Directive('@deprecated(reason: "Use newField")')
+```
+
+- object based - using a JS object to pass the named arguments
 
 ```typescript
-import { Directive, Field, ObjectType } from "type-graphql";
+@Directive("deprecated", { reason: "Use newField" }) // syntax without `@` !!!
+```
 
+The the `@Directive` decorator can be placed over the class property or over the class itself, depending on the needs and the placements supported by the implementation:
```

Comment:
@MichalLytek There is a minor typo here 😉

Reply:
Spend 1 minute looking for the typo before finding "The the" 😄